src/Infrastructure/Repositories/DepartmentRepository.cs

```diff
@@ -0,0 +1,23 @@
+using System.Data;
+using Application.Common.Interfaces;
+using Dapper;
+using Domain.Entities;
+
+namespace Infrastructure.Repositories;
+
+public class DepartmentRepository : IDepartmentRepository
+{
+    private readonly IDbConnection _connection;
+    private readonly IDbTransaction _transaction;
+
+    public DepartmentRepository(IDbConnection connection, IDbTransaction transaction)
+    {
+        _connection = connection;
+    }
+
+    public async Task<Department> GetByIdAsync(Guid id)
+    {
+        var sql = @"SELECT id, name FROM Department WHERE id = @id";
+        return await _connection.QuerySingleOrDefault<Task<Department>>(sql, new { id });
```

Comment:
The result of that method is never a Task

Reply:
what do you mean here, it a Taskl<Department>, we use async from top to bottom no ?